qp_qiime2/qiime2.py

```diff
@@ -434,6 +434,98 @@ def alpha_correlation(qclient, job_id, parameters, out_dir):
                      % (std_out, std_err))
         return False, None, error_msg
 
-    ainfo = [ArtifactInfo('o-visualization', 'q2_visualization',
+    ainfo = [ArtifactInfo('q2_visualization', 'q2_visualization',
                           [(o_visualization, 'qiime2-visualization')])]
     return True, ainfo, ""
+
+
+def taxa_barplot(qclient, job_id, parameters, out_dir):
+    """generate taxa barplot calculations
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to rarefy
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    boolean, list, str
+        The results of the job
+    """
+    out_dir = join(out_dir, 'taxa_barplot')
+    if not exists(out_dir):
+        mkdir(out_dir)
```

Comment:
Can we guarantee that all subdirectories will exist, or should this use `mkdirs` instead?